README.md

```diff
@@ -2102,6 +2103,53 @@ The error message thrown by mailgun is passed to the response object when an ema
 Mailgun error codes can be found
 [here](https://documentation.mailgun.com/en/latest/api-sending.html#error-codes).
 
+## Localization
+
+### Setting application properties
+
+Form flow library will read the languages set in an apps `application.properties`.
+
+```yaml
+# application.yaml
+form-flow:
+  languages: en, es
+```
+
+Our language selector menu button will create links both inside of the button and nav links based on
+the value provided in `form-flow.languages`. If none is given, the default is: `en`.
+
+#### Language tag syntax
+
+Refer to
+the [language tag syntax](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang#language_tag_syntax)
+to see what language you want to support.
+
+### Supplying translation files
+
+Translations need to be added to `messages_xx.properties` files in the `src/main/resources`
+directory, where `xx` is the language the translations are for.
+
+#### Messages in the library
+
+The library comes with [English](src/main/resources/messages-form-flow.properties)
+and [Spanish](src/main/resources/messages-form-flow_es.properties) messages. These load into the
+app messages and get override by message keys that exist in
+the [starter app repository](https://github.com/codeforamerica/form-flow-starter-app/blob/main/src/main/resources).
+
+### How Spring Boot handles localization
+
+Spring boot uses
+a [locale change interceptor](https://www.baeldung.com/spring-boot-internationalization#localechangeinterceptor)
+to intercept a `lang` query parameter (by default) to then change the locale
+and [sets a cookie](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/i18n/CookieLocaleResolver.html)
+to remember the locale.
+
+### Mixpanel event from language selector
+
+If your application is using mixpanel, and you want to keep track clicks into the language selector
```

Comment:
I think it might be worth capitalizing Mixpanel (I think they capitalize it, but unsure). Also, a link to mixpanel.com, might be nice.